02_bubble/runnerdave/main.go

```diff
@@ -0,0 +1,27 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func main() {
+	fmt.Fprint(out, bubble([]int{3, 2, 1, 5}))
+}
+
+func bubble(s []int) []int {
```

Comment:
This function modifies `s`, it shouldn't. I realise you are passing in a copy in your tests, which is probably just as good, but we have to go with the lab instructions and require the function does the copy.

Reply:
good catch 😉